typescript/introduction/tsc-interfaces-classes/tsc-interfaces-intro.md

```diff
@@ -0,0 +1,86 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Introduction to Interfaces
+
+---
+## Content
+
+The main data type that you can use as a collection of values and behaviors is called an *object*.
+
+```ts
+function f(companyObj: { name: string }) {
+  console.log(companyObj.name);
+}
+const someObject = {
+  name: "Enki"
+};
+f(someObject);
+```
+
+The function will verify that `companyObj` has a property called `name` of type `string`. 
+
+```ts
+function f(companyObj: { name: string }) {
+  console.log(companyObj.name);
+}
+const someObject = {
+  name: "Enki",
+  product: "EnkiApp" // ignored without an error
+};
+f(someObject);
+```
+
+It won't care about the order of the properties or if any additional ones exist.
```

Comment:
Now that I think of it, this is not relevant for this particular insight and I think we should remove it.